python/pyspark/sql/tests/test_dataframe.py

```diff
@@ -67,6 +67,35 @@ def test_help_command(self):
         pydoc.render_doc(df.foo)
         pydoc.render_doc(df.take(1))
 
+    # SPARK-36034 test that drop duplicates throws a type error when in correct type provided
+    def test_drop_duplicates(self):
+        schema = StructType([
+            StructField("name", StringType(), True),
+            StructField("age", IntegerType(), True)]
+        )
+
+        # shouldn't drop a non-null row
+        self.assertEqual(self.spark.createDataFrame(
+            [(u'Alice', 50), (u'Alice', 60)], schema).dropDuplicates().count(),
+            2)
+
+        self.assertEqual(self.spark.createDataFrame(
+            [(u'Alice', 50), (u'Alice', 60)], schema).dropDuplicates("name").count(),
+            1)
+
+        self.assertEqual(self.spark.createDataFrame(
+            [(u'Alice', 50), (u'Alice', 60)], schema).dropDuplicates(["name"]).count(),
+            1)
+
+        self.assertEqual(self.spark.createDataFrame(
+            [(u'Alice', 50), (u'Alice', 60)], schema).dropDuplicates(["name", "age"]).count(),
+            2)
+
+        with self.assertRaises(TypeError):
+            self.spark.createDataFrame(
+                [(u'Alice', 50), (u'Alice', 60)], schema
+            ).dropDuplicates("name", "age").count()
```

Comment:
How does this work? `dropDuplicates` only takes one argument. Can you check the exception message together with `assertRaisesRegex`?

Reply:
yep, changed!